packages/backend-common/src/logging/rootLogger.ts

```diff
@@ -31,6 +34,28 @@ export function setRootLogger(newLogger: winston.Logger) {
   rootLogger = newLogger;
 }
 
+/** @public */
+export function setRedactionMap(newRedactionMap: RedactionMap) {
+  redactionMap = newRedactionMap;
+}
+
+/**
+ * A winston formatting function that finds occurrences of filteredKeys
+ * and replaces them with the corresponding identifier.
+ */
+function redactLogLine(info: winston.Logform.TransformableInfo) {
+  // TODO(hhogg): The logger is created before the config is loaded,
+  // because the logger is needed in the config loader. There is a risk of
+  // a secret being logged out during the config loading stage 🤷‍♂️
+  if (redactionMap) {
+    Object.entries(redactionMap || {}).forEach(([key, value]) => {
+      info.message = info.message.replace(new RegExp(key, 'g'), `{{${value}}}`);
```

Comment:
I think we need to optimize this to not see a performance hit tbh. Best only compile a regexes whenever the redaction map changes. I imagine a single regex with `(secret1|secret2|secret3)` will be a lot faster as well

Reply:
I was on the fence about both these, I jumped onto the wrong side 😄  will give this approach a go :+1: